dds/InfoRepo/DCPSInfoRepo.mpc

```diff
@@ -1,4 +1,4 @@
-project(*Lib): dcpslib, iortable, dcps_inforepodiscovery, DCPSInfoRepo_bits, imr_client, svc_utils, install_lib {
+project(*Lib): dcpslib, dcps_tcp, iortable, dcps_inforepodiscovery, DCPSInfoRepo_bits, imr_client, svc_utils, install_lib {
```

Comment:
Why does it need `dcps_tcp` even when built in topics are disabled?

Reply:
I'm pretty sure I was getting a link error to TcpInst::get_local_address() without this.

Reply:
There's a line in DCPSInfo_i.cpp: tcp_inst->local_address(listen_str);

Reply:
Why is that line compiled when built-in topics are disabled?

Reply:
Because there are no #define guards around it. It doesn't look like init_transport method gets called when there's no support for BITs, but this is still producing a dependency on the tcp transport.

So what's the right fix here? Add the guards for all TCP-specific code and then make the MPC dependency upon tcp transport library contingent on BIT support? Or add the guards to the entirety of init_transport's internals and do the same?

Reply:
Oh, I see the DCPSInfoRepo_bits.mpb now... I'll back this out and put in the #defines.

Reply:
Right, the DCPSInfoRepo doesn't need to use any transport implementations when Built-in Topics are excluded at build time.

> MPC dependency upon tcp transport library contingent on BIT support?

This is already done, see `DCPSInfoRepo_bits`